core/files/fileuploader/uploader.go

```diff
@@ -419,23 +419,25 @@ func (u *uploader) Upload(ctx context.Context) (result UploadResult) {
 		opts = append(opts, u.opts...)
 	}
 
-	var addResult *addToStorageResult
+	var addResult *files.AddResult
 	if u.fileType == model.BlockContentFile_Image {
-		addResult, err = u.addImageToStorage(ctx, opts)
+		addResult, err = u.fileService.ImageAdd(ctx, u.spaceId, opts...)
 		if errors.Is(err, image.ErrFormat) || errors.Is(err, mill.ErrFormatSupportNotEnabled) {
 			return u.SetType(model.BlockContentFile_File).Upload(ctx)
 		}
 		if err != nil {
 			return UploadResult{Err: fmt.Errorf("add image to storage: %w", err)}
 		}
 	} else {
-		addResult, err = u.addFileToStorage(ctx, opts)
+		addResult, err = u.fileService.FileAdd(ctx, u.spaceId, opts...)
 		if err != nil {
 			return UploadResult{Err: fmt.Errorf("add file to storage: %w", err)}
 		}
 	}
-	result.MIME = addResult.mime
-	result.Size = addResult.size
+	defer addResult.Commit()
```

Comment:
We need to use transaction to prevent creation of multiple objects for the same file